packages/ic-management/src/utils/transform.utils.ts

```diff
@@ -0,0 +1,30 @@
+import type { ActorConfig, CallConfig } from "@dfinity/agent";
+import { Principal } from "@dfinity/principal";
+
+type CallTransform = Required<ActorConfig>["callTransform"];
+
+type QueryTransform = Required<ActorConfig>["queryTransform"];
+
+/**
+ * Transformer function for service creation with `callTransform` or `queryTransform`.
+ *
+ * This function maps the `effective_canister_id` for calls to the Management Canister (`aaaaa-aa`).
+ *
+ * Original source `getManagementCanister` in agent-js.
+ *
+ * Providing a transformer is required to determine the effective_canister_id when the request is an update call to the Management Canister (aaaaa-aa).
+ *
+ * @link https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-effective-canister-id
+ **/
+export const transform: CallTransform | QueryTransform = (
+  _methodName: string,
+  args: unknown[],
+  _callConfig: CallConfig,
+): { effectiveCanisterId: Principal } => {
+  const first = args[0] as { canister_id: string };
+  let effectiveCanisterId = Principal.fromHex("");
+  if (first && typeof first === "object" && first.canister_id) {
+    effectiveCanisterId = Principal.from(first.canister_id as unknown);
+  }
```

Comment:
I don't know if we use a functional approach in `ic-js`, but this could be...?

```typescript
const effectiveCanisterId = first && typeof first === "object" && first.canister_id ? Principal.from(first.canister_id as unknown) : Principal.fromHex("")
```

Reply:
I'll provide another PR to refactor the logic within the function, didn't wanted to change it in this PR which is closer to a refactoring. Thanks for the feedback!